presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushAggregationThroughOuterJoin.java

```diff
@@ -272,6 +295,12 @@ private MappedAggregationInfo createAggregationOverNull(AggregationNode referenc
         for (Map.Entry<Symbol, AggregationNode.Aggregation> entry : referenceAggregation.getAggregations().entrySet()) {
             Symbol aggregationSymbol = entry.getKey();
             AggregationNode.Aggregation aggregation = entry.getValue();
+
+            boolean symbolDependencyCorrect = checkSymbolDependency(aggregation, sourcesSymbolMapping);
+            if(!symbolDependencyCorrect) {
+                return null;
```

Comment:
please return `Optional.empty()`, unless you have a really good reason to use `null` directly which can be only true in execution.